### ISSUE TITLE
Error when using fzf-tab's preview feature

### ISSUE THREAD

**Issue by NathanGrenier** [Aloxaf/fzf-tab#444]
**Describe your question**
- Windows 10
- zsh
- installed with zinit

Getting a weird error when using the preview feature in zsh. Not quite sure what to do to resolve it.

This is my .zshrc file:
```bash
# Enable Powerlevel10k instant prompt. Should stay close to the top of ~/.zshrc.
# Initialization code that may require console input (password prompts, [y/n]
# confirmations, etc.) must go above this block; everything else may go below.
if [[ -r "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh" ]]; then
  source "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh"
fi

# Set the directory we want to store zinit and plugins
ZINIT_HOME="${XDG_DATA_HOME:-${HOME}/.local/share}/zinit/zinit.git"

# Download Zinit, if it's not there yet
if [ ! -d "$ZINIT_HOME" ]; then
        mkdir -p "$(dirname $ZINIT_HOME)"
        git clone https://github.com/zdharma-continuum/zinit.git "$ZINIT_HOME"
fi

# Source/Load zinit
source "${ZINIT_HOME}/zinit.zsh"

# Add Powerlevel10k
zinit ice depth=1; zinit light romkatv/powerlevel10k

# To customize prompt, run `p10k configure` or edit ~/.p10k.zsh.
[[ ! -f ~/.p10k.zsh ]] || source ~/.p10k.zsh

# Install zsh plugins
zinit light Aloxaf/fzf-tab
zinit light zsh-users/zsh-syntax-highlighting
zinit light zsh-users/zsh-completions
zinit light zsh-users/zsh-autosuggestions

# Add in snippets
zinit snippet OMZP::git

# Load completions
autoload -U compinit && compinit

zinit cdreplay -q

# History
HISTSIZE=5000
HISTFILE=~/.zsh_history
SAVEHIST=$HISTSIZE
HISTDUP=erase
setopt appendhistory
setopt sharehistory
setopt hist_ignore_space
setopt hist_ignore_all_dups
setopt hist_save_no_dups
setopt hist_ignore_dups
setopt hist_find_no_dups

# Completion styling
zstyle ':completion:*' matcher-list 'm:{a-z}={A-Za-z}'
zstyle ':completion:*' list-colors "${(s.:.)LS_COLORS}"
zstyle ':completion:*' menu no
zstyle ':fzf-tab:complete:cd:*' fzf-preview 'ls --color $realpath'
zstyle ':fzf-tab:complete:__zoxide_z:*' fzf-preview 'ls --color $realpath'

# keybindings
bindkey '^f' autosuggest-accept
bindkey '^p' history-search-backward
bindkey '^n' history-search-forward

# Aliases
alias cl='clear'
alias ls='ls --color'

# Shell integrations
[ -f ~/.fzf.zsh ] && source ~/.fzf.zsh
eval "$(zoxide init zsh)"

# Navigate to home directory
cd ~
```

## Screenshot of the Error
![image](https://github.com/Aloxaf/fzf-tab/assets/56941028/2c6cab7c-41ab-42cf-861c-c1361ae8723e)

**Comment by Aloxaf** [comment]
what's the value of `$ZSH_NAME` on windows?

**Comment by jathek** [comment]
just "zsh"

![image](https://github.com/Aloxaf/fzf-tab/assets/692914/acd230a7-c1fa-4d53-94eb-ff76a41c3f4e)

**Comment by srouquette** [comment]
I kinda have the same problem, but it's most likely related to msys2.  
If I run this simple command, I have the preview: `fzf --preview='less {}'`  
But if I run with fzf-tab, I have this error:

![image](https://github.com/user-attachments/assets/f1074c7d-bcfc-4a83-a34f-d93dcf09e3e7)

(it tries to run zsh as if it was in the current dir)

If I fix the path to zsh, I have this new error: 

![image](https://github.com/user-attachments/assets/b8057885-ddec-452a-b16b-0367fd6b6cb4)

It seems related to path resolution with the temp dir ?

**Comment by jathek** [comment]
I tried your zsh path fix and get the same (new) error as you. I agree there's a second issue with path resolution. $TEMP on Windows (with msys2 or git-for-windows) is `/tmp`, but behind the scenes it is actually a Windows path of `C:\Users\USERNAME\AppData\Local\Temp`. The Windows path is being called by zsh at some point (asked by fzf?) and the `\` are being interpreted as escape characters, so the path becomes invalid.

I did check my temp folder, and a folder is actually being created in the temp folder and populated with files by fzf-tab.
![image](https://github.com/user-attachments/assets/0ac01338-6d25-4343-a59d-371f08e218f5)

However, the `/tmp/fzf-temp-123456789` temp files (from fzf?) aren't created anymore because of the path issue. If I remove the ZSH path fix, the `/tmp/fzf-temp-123456789` files get created, but the zsh "file does not exist" error returns.

(the above folder from fzf-tab only gets created properly after the following change in -ftb-fzf. Without it, the folder is "zsh-fzf-tab-" instead of "zsh-fzf-tab-USERNAME")
```diff
- local tmp_dir=${TMPPREFIX:-/tmp/zsh}-fzf-tab-$USER
+ local tmp_dir=${TMPPREFIX:-/tmp/zsh}-fzf-tab-$(whoami)
```

**Comment by srouquette** [comment]
I checked fzf code:

```go
func WriteTemporaryFile(data []string, printSep string) string {
	f, err := os.CreateTemp("", "fzf-temp-*")
	if err != nil {
		// Unable to create temporary file
		// FIXME: Should we terminate the program?
		return ""
	}
```

🤣

**Comment by xd003** [comment]
Can confirm that its not working on windows msys2 zsh, works fine for me on linux
Although i don't get any error as in the screenshot posted by the OP, previews just don't show up

**Comment by sylvain-rouquette** [comment]
I can have the preview if I'm using fzf + other app. The problem seems to come from fzf-tab.

[my zdotdir](https://github.com/srouquette/zdotdir). I have a preview with CTRL+T and ALT+C.

**Comment by xd003** [comment]
> I can have the preview if I'm using fzf + other app. The problem seems to come from fzf-tab.

Just curious what is the alternative that you are talking about

**Comment by srouquette** [comment]
the default keybindings are already quite good. I added preview on CTRL+T and ALT+C. You can also autocomplete cd with `cd **<TAB>`. I'm just missing the completion for command line parameters.

I'm using the [OMZ plugin](https://github.com/srouquette/zdotdir/blob/main/.zplugins#L36), and [how I customized it](https://github.com/srouquette/zdotdir/blob/main/conf.d/fzf.zsh) with fd, eza and rg.

**Comment by RedCommand** [comment]
Did someone find a fix?

**Comment by Mantene** [comment]
I was able to get this working by following this the instruction in #295 and just running this in the fzf-tab plugin directory - git checkout 703ab

**Comment by jathek** [comment]
The fix in 703ab22 was reversed in the next commit ca0dee3. It looks like the Windows fix broke a keybinding for something else. But the Windows thing broke again 😦

**Comment by Mantene** [comment]
Yes. So I pulled the older commit where the fix was there and that works. Sure, it isn't the latest and greatest, but it works just fine.
